src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,498 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+                    )}
+                    {vm.persistent &&
+                    <PermanentChange idPrefix={idPrefix}
+                                     permanent={diskParams.permanent}
+                                     onValueChanged={onValueChanged}
+                                     vm={vm} />}
+                </Form>
+                <AdditionalOptions cacheMode={diskParams.cacheMode}
+                                   device={diskParams.device}
+                                   idPrefix={idPrefix}
+                                   onValueChanged={onValueChanged}
+                                   busType={diskParams.busType}
+                                   serial={diskParams.serial}
+                                   validationFailed={validationFailed}
+                                   supportedDiskBusTypes={supportedDiskBusTypes} />
+            </>
+        );
     }
 
-    onInsertClicked() {
-        const Dialogs = this.context;
-        const close = Dialogs.close;
-        const { vm } = this.state;
-        const { disk } = this.props;
+    return (
+        <Modal position="top" variant="medium" id={`${idPrefix}-dialog-modal-window`} isOpen onClose={Dialogs.close}
+               title={isMediaInsertion ? _("Insert disc media") : _("Add disk")}
+               footer={
+                   <AddDiskModalFooter
+                     dialogLoading={dialogLoading}
+                     dialogErrorSet={dialogErrorSet}
+                     diskParams={diskParams}
+                     idPrefix={idPrefix}
+                     isMediaInsertion={isMediaInsertion}
+                     mode={mode}
+                     setValidate={setValidate}
+                     storagePoolName={storagePoolName}
+                     storagePools={storagePools}
+                     validate={validate}
+                     validationFailed={_validationFailed}
+                     verificationInProgress={verificationInProgress}
+                     vm={vm}
+                     vms={vms}
+                   />
+               }
+        >
+            {dialogError && <ModalError dialogError={dialogError} dialogErrorDetail={dialogErrorDetail} />}
+            {defaultBody}
+        </Modal>
+    );
+};
 
-        const validation = this.validateParams();
-        if (Object.getOwnPropertyNames(validation).length > 0)
-            return this.setState({ addDiskInProgress: false, validate: true });
+const AddDiskModalFooter = ({
+    dialogErrorSet,
+    dialogLoading,
+    diskParams,
+    idPrefix,
+    isMediaInsertion,
+    mode,
+    setValidate,
+    storagePoolName,
+    storagePools,
+    validate,
+    validationFailed,
+    verificationInProgress,
+    vm,
+    vms,
+}) => {
+    const [addDiskInProgress, setAddDiskInProgress] = useState(false);
+    const Dialogs = useDialogs();
 
-        this.setState({ addDiskInProgress: true, validate: false });
+    const onInsertClicked = () => {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231117-134447-13cb53e1-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)